### ISSUE TITLE
Go to constructor doesn't work

### ISSUE THREAD

**Issue by Yanpas** [MaskRay/ccls#208]
Example 1:
```cpp
#include <string>
// template <typename T = int>
struct Bar {
    Bar(int) {

    }
};
using std::string;
void x(){
    Bar string(1);
    Bar bbb(1);
}
```

Place cursor on string variable and ctrl-click (goto definition IDK) and you are jumped to `struct Bar {`
Place cursor on bbb variable and ctrl-click (goto definition IDK) and you are jumped to `struct Bar {` and you either jump to constuctor (rarely) either struct Bar (same as in the first case).

**Comment by MaskRay** [comment]
I don't quite understand the last sentence. Can you make it clearer?

References of constructors have wider ranges (you can `textDocument/definition` on ` string(`)
https://github.com/MaskRay/ccls/wiki/FAQ#definitions

**Comment by Yanpas** [comment]
Let me rephrase:

The problem is that goto definition is undeterministic. If we create class instance this way on the stack `Klass instance(arg1, arg2);` then after going to definition of `instance` I often go to Klass definition, but sometimes I go to Klass::Klass constructor definition. I should go to constructor always.

It seems to me the ccls has the following logic: It searcher in it's index the best possible option, if it doesn't find the correct option in some short time period it falls back to something that is somehow similar.

In the main project I face it the same way: there is variable called future (`ns1::Klass1 future;`) and class called future in another namespace (`ns2::future`). If I ctrl-click on future variable I go to future class which is definetely wrong.

**Comment by MaskRay** [comment]
See my edited comment https://github.com/MaskRay/ccls/issues/208#issuecomment-454622615

**Comment by Yanpas** [comment]
> one of its definitions is returned instead

Maybe actual constructor is more useful than type definition?
